build_tools/scripts/lint.sh

```diff
@@ -54,9 +54,8 @@ echo "***** buildifier *****"
 ${scripts_dir}/run_buildifier.sh
 git diff --exit-code
 
-echo "***** yapf *****"
-# Don't fail script if condition is false
-git diff -U0 main | ./third_party/format_diff/format_diff.py yapf -i
```

Comment:
Huh... looks like we could remove https://github.com/openxla/iree/tree/main/third_party/format_diff in favor of https://github.com/llvm/llvm-project/blob/main/clang/tools/clang-format/clang-format-diff.py (our version was forked to add yapf alongside clang)